### DISCUSSION TITLE
How do I create a mesh with a different material for each polygon?

### DISCUSSION THREAD

**Discussion by H-A-M-G-E-R** [bevyengine/bevy#11005] [Q&A]
example: cube with a different color for each side

**Comment by stepancheg** [comment]
Mesh does not have material attached to it. It only contains vertices. So it is not possible to create a mesh even with one material.

You probably want to just create different meshes for each polygon.

Note you can instantiate them with the same root entity, so for example, transformations need to be applied only to top entity.

Alternatively you can create a texture to map different colors to different sides. Like in [this example](https://bevyengine.org/examples/3D%20Rendering/3d-shapes/).

Or even create a shader which would generate colors.

But the easiest is just to create different entities for different sides.
